### PR TITLE
fix: pagination and filtering issues in `/addresses/:hash/nft`

### LINKED ISSUE #13166: `/api/v2/addresses/:hash/nft` ignores type filter and also returns malformed `next_page_params`
Filtering by type=ERC-404,ERC-1155 still returns ERC-721 tokens. Pagination metadata (next_page_params.token_type) also shows ERC-721 while original filter requests ERC-404,ERC-1155.

```
curl 'https://eth-sepolia.k8s-dev.blockscout.com/api/v2/addresses/0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045/nft?type=ERC-404,ERC-1155'
```

Actual (excerpt)

.items[].token_type -> all "ERC-721"

next_page_params:

```
{
  "type": "ERC-404,ERC-1155",
  "token_type": "ERC-721",
  "token_contract_address_hash": "0x201f25a825e8db36b3a7451849a561307f71565b",
  "token_id": "962",
  "items_count": 50
}
```

### PR DESCRIPTION
**Pull request by fedor-ivn**
Closes #13166.

## Changelog

- Fix issue when multi-type filter (e.g `ERC-404,ERC-1155`) also included not requested types
- Unified NFT pagination across ERC-1155, ERC-404, and ERC-721.
- Resolved edge cases causing sparse pages or skipped types during NFT pagination.

## Checklist for your Pull Request (PR)

- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
- [ ] I checked whether I should update the docs and did so by submitting a PR to [docs repository](https://github.com/blockscout/docs).
- [ ] If I added/changed/removed ENV var, I submitted a PR to [docs repository](https://github.com/blockscout/docs) to update the list of [env vars](https://github.com/blockscout/docs/blob/master/setup/env-variables/README.md) and I updated the version to `master` in the Version column. If I removed variable, I added it to [Deprecated ENV Variables](https://github.com/blockscout/docs/blob/master/setup/env-variables/deprecated-env-variables/README.md) page. After merging docs PR, changes will be reflected in these [pages](https://docs.blockscout.com/setup/env-variables).
- [ ] If I added new DB indices, I checked, that they are not redundant, with PGHero or other tools.
- [ ] If I added/removed chain type, I modified the Github CI matrix and PR labels accordingly.

## Summary by CodeRabbit

- New Features
  - Unified NFT pagination across ERC-1155, ERC-404, and ERC-721 so pages fill by continuing across types when one is exhausted.
  - Exposed a canonical list of allowed NFT type labels.
  - Multi-type filters support comma-separated types and respect requested type sets.

- Bug Fixes
  - Pagination ordering adjusted for predictable type sequencing.
  - Next-page links no longer leak the original composite type filter.

- Tests
  - Added comprehensive multi-type pagination and filtering tests for NFT endpoints.

### CODE REVIEW

File: apps/explorer/lib/explorer/chain/token/instance.ex [hunk 1]
```diff
@@ -136,46 +136,93 @@ defmodule Explorer.Chain.Token.Instance do
...
+              page_size: page_size - count
+            })
+          end
+
+        fetched = fetch_by_type_and_address_hash(type, address_hash, options)
+        new_list = list ++ fetched
+        new_count = count + length(fetched)
+
+        if new_count >= page_size do
+          {:halt, {Enum.take(new_list, page_size), page_size}}
```
[the remark this answers is not in the document]
Comment (fedor-ivn) [reply apps/explorer/lib/explorer/chain/token/instance.ex:180]: I think this is logically invalid. The size of a list returned should be equal to the counter (second element of a tuple)

File: apps/explorer/lib/explorer/chain/token/instance.ex [hunk 2]
```diff
@@ -136,46 +136,93 @@ defmodule Explorer.Chain.Token.Instance do
...
-        erc_721 = erc_721_token_instances_by_owner_address_hash(address_hash, options)
-
-        if length(erc_721) == paging_options.page_size do
-          erc_721
-        else
-          erc_1155 = erc_1155_token_instances_by_address_hash(address_hash, options)
-          erc_404 = erc_404_token_instances_by_address_hash(address_hash, options)
+        token_types
+    end
+  end
```
[the remark this answers is not in the document]
Comment (fedor-ivn) [reply apps/explorer/lib/explorer/chain/token/instance.ex:217]: This preprocessing is done in the controller (when actually parsing token types from query params)

File: apps/explorer/lib/explorer/chain/token/instance.ex [hunk 3]
```diff
@@ -136,46 +136,93 @@ defmodule Explorer.Chain.Token.Instance do
...
+        if new_count >= page_size do
+          {:halt, {Enum.take(new_list, page_size), page_size}}
+        else
+          {:cont, {new_list, new_count}}
+        end
+      end)
 
-  defp nft_list(address_hash, ["ERC-404"], options) do
-    erc_404_token_instances_by_address_hash(address_hash, options)
+    acc
```
[the remark this answers is not in the document]
Comment (fedor-ivn) [reply apps/explorer/lib/explorer/chain/token/instance.ex:186]: Propose diff pls

File: apps/explorer/lib/explorer/chain/token/instance.ex (hunk 2 above)
[the remark this answers is not in the document]
Comment (fedor-ivn) [reply apps/explorer/lib/explorer/chain/token/instance.ex:217]: > In this case, Enum.drop_while(["ERC-404"], &(&1 != "ERC-721")) would drop all elements and return [], causing no results to be returned even though there might be ERC-404 tokens.

This seems to be logical, no?

File: apps/explorer/lib/explorer/chain/token/instance.ex [hunk 4]
```diff
@@ -136,46 +136,93 @@ defmodule Explorer.Chain.Token.Instance do
...
+          if Enum.empty?(list) do
+            options
+          else
+            Keyword.put(options, :paging_options, %PagingOptions{
+              page_size: page_size - count
+            })
+          end
+
+        fetched = fetch_by_type_and_address_hash(type, address_hash, options)
+        new_list = list ++ fetched
```
[the remark this answers is not in the document]
Comment (fedor-ivn) [reply apps/explorer/lib/explorer/chain/token/instance.ex:176]: Can you propose a diff that implements this approach?

**Review by vbaranov — APPROVED** [review]
